connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -1112,9 +1117,7 @@ public HerderRequest(long at, Callable<Void> action, Callback<Void> callback) {
 
         @Override
         public int compareTo(HerderRequest o) {
-            final int soonest = Long.compare(at, o.at);
-            // If tied, returning a positive value should respect insertion order.
-            return soonest != 0 ? soonest : 1;
+            return Long.compare(id, o.id);
```

Comment:
The priority queue defines priorities primarily based on user defined delays (`addRequest` takes a `delay` as argument. The sequence number should break ties between requests that have the same priority, which is a specific time in the future). 

Because the priorities depend on user defined delays, such priorities might correspond to the creation time of a request, but this is not required. Thus, `compareTo` needs first to compare deadlines, and if they are equal use the sequence number to break ties.